src/metabase/driver/sql/query_processor.clj

```diff
@@ -1282,13 +1282,15 @@
 ;;  aggregation REFERENCE e.g. the ["aggregation" 0] fields we allow in order-by
 (defmethod ->honeysql [:sql :aggregation]
   [driver [_ index]]
-  (driver-api/match-one
-    (nth (:aggregation *inner-query*) index)
+  (driver-api/match-one (nth (:aggregation *inner-query*) index)
+    [:aggregation-options ag (options :guard driver-api/qp.add.source-alias)]
+    (->honeysql driver (h2x/identifier :field-alias (driver-api/qp.add.source-alias options)))
```

Comment:
Drivers are supposed to use the `driver-api/qp.add.*` keys when generating native queries, not stuff like `:name` (which is not necessarily deduplicated/escaped/etc. so this was the root cause of the bug